assets/extensions/card.js

```diff
@@ -11,26 +16,50 @@ import ExtensionActions from './extension-actions';
 /**
  * Extensions card component.
  *
- * @param {Object}  props           Component props.
- * @param {boolean} props.extension Extension object.
+ * @param {Object}   props             Component props.
+ * @param {string}   props.title       Card title (extension title will be used as fallback).
+ * @param {string}   props.excerpt     Card excerpt (extension excerpt will be used as fallback).
+ * @param {string}   props.badgeLabel  Badge label (will check extension update if it's not defined).
+ * @param {Object[]} props.customLinks Array with custom links containing the link props.
+ * @param {Object}   props.extension   Extension object.
+ * @param {Object}   props.htmlProps   Wrapper extra props.
  */
-const Card = ( { extension } ) => (
-	<article className="sensei-extensions__card">
+const Card = ( {
+	title,
+	excerpt,
+	badgeLabel,
+	customLinks,
+	extension,
```

Comment:
WDYT about removing the `extension` parameter, and just using this as:

```jsx
<Card { ...extension } { ...cardProps } />
```

A higher-level `ExtensionCard` could wrap this, taking care of extension-specific things, like setting the badge and building the action buttons (passing them in an `actions` props that works like `customLinks`, but can support any component).

The general concept would be that the same props of a Card could come from either an extension, from the API, or mixed.

Reply:
Good idea! Refactored here: https://github.com/Automattic/sensei/pull/4197/commits/a20b0548881936f1172e82127e5fff9bf947a9f7

I just didn't create the `ExtensionCard`, because it seems simpler like it's currently after this refactor. WDYT?